ts/test/mock_transpiler.bzl

```diff
@@ -1,26 +1,68 @@
 "Fixture to demonstrate a custom transpiler for ts_project"
 
-load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
-load("@bazel_skylib//rules:write_file.bzl", "write_file")
-
 _DUMMY_SOURCEMAP = """{"version":3,"sources":["%s"],"mappings":"AAAO,KAAK,CAAC","file":"in.js","sourcesContent":["fake"]}"""
 
-def mock(name, srcs, js_outs, map_outs, **kwargs):
-    """Mock transpiler macro.
+def _mock_impl(ctx):
+    src_files = [src for src in ctx.files.srcs if src.short_path.endswith(".ts") and not src.short_path.endswith(".d.ts")]
+    out_files = []
+
+    for src in src_files:
+        out_path = src.short_path[len(ctx.attr.pkg_dir) + 1:] if src.short_path.startswith(ctx.attr.pkg_dir) else src.short_path
```

Comment:
I don't understand why this is need (and the `pkg_dir` to do it). Without this it is nesting the output files too much, with the directory name repeated such as `test/test/ts/test/file.js`.

I don't see where in the default tsc this is avoided..? 🤔

Reply:
maybe we should figure that out now then? I added a commit that uses the ts_lib logic.

Reply:
I was looking at those but unsure what ts_lib was doing different 🤷 

I guess this is an internal test so there's no issue with using internal utils (ts_lib)?

Reply:
Yeah and ts_lib could be more public, others will want to do it this way I think